### DISCUSSION TITLE
Automatically cleanup old build files

### DISCUSSION THREAD

**Discussion by akiran** [remix-run/remix#2425] [Proposals]
As remix generates new build files with new hashes on every code change without deleting old files, build folder size is increasing over time and consuming huge hard disk space.

Currently, the build folder of my remix project crossed 24 GB.  I had to manually check and delete the build folder when I found that my computer was running out of disk space.

It would be great if remix could automatically clean the old and unused build files and keep the build folder size small.
Let me know if this is possible.

**Comment by sergiodxa** [comment]
This was added last year https://github.com/remix-run/remix/pull/252

**Reply by akiran** [reply]
I am still having large build folders. How to invoke cleanup?

**Reply by sergiodxa** [reply]
It should happen automatically when you stop the dev server (`remix dev` or `remix watch`).

If it's not happening for you, then it may not be working and that's a bug.

**Reply by akiran** [reply]
It's not happening for me.

Should I open a new issue and report this as a bug?

**Reply by daniel-a-marshall** [reply]
I'm having the same problem as well.

**Reply by thomasverleye** [reply]
Same, see https://github.com/remix-run/remix/discussions/3156

**Comment by HerrBertling** [comment]
This seems to be the case for my Remix setup (using `1.2.2` with Architect currently) for the `remix build` step. Just noticed because I could not deploy to AWS (separate CDK workflow instead of Architect because of reasons…). So I checked the `/public/build/` folder and it was packed with old stuff. Shouldn't it be wiped as well before running a new build?

**Comment by devbytyler** [comment]
This was also happening to me. My public/build/ had somehow gotten to 300Mb after zipping. When I deployed to Elastic Beanstalk, it was failing due to large size. I deleted the build directory, rebuilt and zipped and the new size was 3Mb.

I'm runnning:
    "@remix-run/node": "^1.3.3",
    "@remix-run/react": "^1.3.3",
    "@remix-run/serve": "^1.3.3",
(Remix App-server)

**Comment by bpeter340** [comment]
I'm running into this issue as well. My public/build folder is currently consuming 60GB of storage space. I only noticed this because I got a notification that my computer was running low on storage and I found that to be odd.

I'm running:
"@remix-run/node": "^1.8.2",
"@remix-run/react": "^1.8.2",
"@remix-run/serve": "^1.8.2",
